sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/SdkMeterProvider.java

```diff
@@ -153,14 +153,15 @@ public void close() {
       if (!disableSynchronousCollection) {
         lastCollectionTimestamp.lazySet(currentNanoTime);
       }
+      CollectionInfo info = collectionInfoMap.get(handle);
+      if (info == null) {
+        return Collections.emptyList();
```

Comment:
This should likely through some kind of logic error bug.  We own both the Handle-creation *and* the collectionInfoMap, so we should never have a value that's not in the map, if we do the SDK itself has a bug.

Reply:
Made it an exception - looked through and it looks like the map is a side channel to workaround a circular dependency between reader and collectioninfo. We can probably improve in the future by refactoring to have the reader's info (temporality settings) passed separately from the collectioninfo